### DISCUSSION TITLE
Intellisense not working for pymupdf package only on vscode

### DISCUSSION THREAD

**Discussion by jim2mul** [pymupdf/PyMuPDF#1898] [Looking for help]
I am using the latest version of pymupdf (1.20.2) on a Windows machine with VSCode. Intellisense is working with all other packages. I have checked the troubleshooting steps listed here. -> https://donjayamanne.github.io/pythonVSCodeDocs/docs/troubleshooting_intellisense/ but its still not working.

Has anyone else encountered this issue and managed to resolve?

**Comment by Augus1999** [comment]
What kind of problems have you encountered? I also use VSCode on Windows but do not have any problems.🤔

**Reply by cobaltautomationdev** [reply]
Me too, auto Intellisense not work for pymupdf in vscode. how to fix it?

**Comment by ivanstepanovftw** [comment]
I am importing fitz as usual, but instead of using `fitz.open`, I am using `fitz.Document`, that is defined at `fitz.fitz.Document`, because Intellisense works with it.

Same for everything else: instead of `page.get_text("dict")` I am using `fitz.utils.get_text(page, "dict")`.

**Comment by JorjMcKie** [comment]
yes, it looks like VS code does not  look at `__init__.py` where all these definitions are being made ...

**Comment by JorjMcKie** [comment]
On the other hand: how much effort do you want to sacrifice for the shortcomings of an Intellisense system ...

**Reply by cobaltautomationdev** [reply]
for me , I appreciate the benefits of the Intellisense system, but I also value my own skills and judgment. I am willing to invest some effort to overcome the challenges of the system, but not at the expense of my productivity and creativity.
